cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerObjectStorageValidator.java

```diff
@@ -67,31 +69,33 @@ public ValidationResult validateObjectStorage(AmazonIdentityManagementClient iam
                 } else if (CloudIdentityType.LOG.equals(cloudIdentityType)) {
-                    validateLocation(iam, instanceProfile, cloudFileSystem, logsLocationBase, backupLocationBase, accountId, resultBuilder);
+                    validateLocation(iam, instanceProfile, cloudFileSystem, logsLocationBase, backupLocationBase, accountId, backupOperationType, resultBuilder);
                 }
             }
         }
         return resultBuilder.build();
     }
 
     private void validateIDBroker(AmazonIdentityManagementClient iam, InstanceProfile instanceProfile,
-            CloudS3View cloudFileSystem, String backupLocationBase, String accountId, ValidationResultBuilder resultBuilder) {
+            CloudS3View cloudFileSystem, String backupLocationBase, String accountId,
+            BackupOperationType backupOperationType, ValidationResultBuilder resultBuilder) {
         LOGGER.info("Permission validation on IBBroker role: {}", instanceProfile.getInstanceProfileName());
         awsInstanceProfileEC2TrustValidator.isTrusted(instanceProfile, cloudFileSystem.getCloudIdentityType(), resultBuilder);
         Set<Role> allMappedRoles = getAllMappedRoles(iam, cloudFileSystem, resultBuilder);
         awsIDBrokerAssumeRoleValidator.canAssumeRoles(iam, instanceProfile, allMappedRoles, resultBuilder);
-        awsDataAccessRolePermissionValidator.validate(iam, cloudFileSystem, backupLocationBase, accountId, resultBuilder);
-        awsRangerAuditRolePermissionValidator.validate(iam, cloudFileSystem, backupLocationBase, accountId, resultBuilder);
+        awsDataAccessRolePermissionValidator.validate(iam, cloudFileSystem, backupLocationBase, accountId, backupOperationType, resultBuilder);
+        awsRangerAuditRolePermissionValidator.validate(iam, cloudFileSystem, backupLocationBase, accountId, backupOperationType, resultBuilder);
     }
 
     private void validateLocation(AmazonIdentityManagementClient iam, InstanceProfile instanceProfile, CloudS3View cloudFileSystem,
-            String logsLocationBase, String backupLocationBase, String accountId, ValidationResultBuilder resultBuilder) {
+            String logsLocationBase, String backupLocationBase, String accountId,
+            BackupOperationType backupOperationType, ValidationResultBuilder resultBuilder) {
         LOGGER.info("Permission validation on Role: {}", instanceProfile.getInstanceProfileName());
         awsInstanceProfileEC2TrustValidator.isTrusted(instanceProfile, cloudFileSystem.getCloudIdentityType(), resultBuilder);
         awsLogRolePermissionValidator.validateLog(iam, instanceProfile, cloudFileSystem, logsLocationBase, resultBuilder);
-        if (entitlementService.isDatalakeBackupRestorePrechecksEnabled(accountId)) {
+        if (entitlementService.isDatalakeBackupRestorePrechecksEnabled(accountId) && backupOperationType.equals(BackupOperationType.RESTORE)) {
```

Comment:
It looks like we only check the actual value here for AWS and only if it's a restore only. 

Is that correct that for AWS there isn't a lesser permission required for RESTORE?
And that for other cloud providers there is no "Backup only" permission?

Reply:
LOG role needs permissions only while restoring the datalake. This code skips checking the permissions for the log role if the operation is BACKUP.


https://docs.cloudera.com/management-console/cloud/data-lakes/topics/mc-data-lake-dr-restore-dl-content.html


In case of AZURE, there is no difference in permissions needed for BACKUP Vs Restore.